libs/ngxtension/inject-query-params/src/inject-query-params.spec.ts

```diff
@@ -22,22 +35,96 @@ describe(injectQueryParams.name, () => {
 
 		expect(instance.queryParams()).toEqual({ query: 'Angular' });
 		expect(instance.searchParam()).toEqual('Angular');
+		expect(instance.idParam()).toEqual(null);
 		expect(instance.paramKeysList()).toEqual(['query']);
 
 		await harness.navigateByUrl('/search?query=IsCool!&id=2');
 
 		expect(instance.queryParams()).toEqual({ query: 'IsCool!', id: '2' });
+		expect(instance.idParam()).toEqual(2);
 		expect(instance.searchParam()).toEqual('IsCool!');
 		expect(instance.paramKeysList()).toEqual(['query', 'id']);
 	});
+
+	it('returns a signal everytime the query params change with number', async () => {
+		TestBed.configureTestingModule({
+			providers: [
+				provideRouter([{ path: 'search', component: SearchComponent }]),
+			],
+		});
+
+		const harness = await RouterTestingHarness.create();
+
+		const instance = await harness.navigateByUrl(
+			'/search?id=Angular',
+			SearchComponent,
+		);
+
+		expect(instance.queryParams()).toEqual({ id: 'Angular' });
+		expect(instance.idParam()).toEqual(NaN);
+		expect(instance.paramKeysList()).toEqual(['id']);
+
+		await harness.navigateByUrl('/search?&id=2.2');
+
+		expect(instance.queryParams()).toEqual({ id: '2.2' });
+		expect(instance.idParam()).toEqual(2.2);
+		expect(instance.paramKeysList()).toEqual(['id']);
+	});
 });
 
-@Component({
-	standalone: true,
-	template: ``,
-})
-export class SearchComponent {
-	queryParams = injectQueryParams();
-	searchParam = injectQueryParams('query');
-	paramKeysList = injectQueryParams((params) => Object.keys(params));
```

Comment:
Question: Is `queryParams.multiple` a simpler variant of `injectQueryParams((params) => Object.keys(params))`?

Reply:
No, its more like if you know you want all query params of a specific key as an array!

For example you've got the url `/search?productId=1&productId=2`.
Now you want the productIds as an array. With the current implementation `#idParam = injectQueryParams('productId')` you get the type `string|null` even when it should be `string[]|null`. You have no way to tell the method you want an array.

Perhaps `injectQueryParams.array()` would be a better name?